unstructured/metrics/evaluate.py

```diff
@@ -212,9 +235,20 @@ def _write_to_file(dir: str, filename: str, df: pd.DataFrame, mode: str = "w"):
         df["count"] = df["count"].astype(int)
     if "filename" in df.columns and "connector" in df.columns:
         df.sort_values(by=["connector", "filename"], inplace=True)
+    if not overwrite:
+        filename = _uniquity_file(dir, filename)
     df.to_csv(os.path.join(dir, filename), sep="\t", mode=mode, index=False, header=(mode == "w"))
 
 
+def _uniquity_file(dir, filename):
+    counter = 1
+    original_filename, extension = filename.rsplit(".", 1)
+    while os.path.exists(os.path.join(dir, filename)):
+        filename = original_filename + " (" + str(counter) + ")." + extension
+        counter += 1
```

Comment:
I would suggest to use `os.listdir` and filtering to find all the files match the pattern and get the counter values and get max -> increment by 1
Reason:
- repeatedly calling os.path.join, path.exists is more expensive than just calling once listdir
- a while loop is a dangerous trap for infinite loop and I'd suggest we avoid it as much as we can